pkg/apis/core/validation/shoot.go

```diff
@@ -285,7 +285,7 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 
 	if newObjectMeta.DeletionTimestamp != nil && !apiequality.Semantic.DeepEqual(newSpec, oldSpec) {
 		if diff := deep.Equal(newSpec, oldSpec); diff != nil {
-			return field.ErrorList{field.Forbidden(fldPath, strings.Join(diff, ","))}
+			return field.ErrorList{field.Forbidden(fldPath, fmt.Sprintf("can't edit shoot while it's in deletion. Requested changes: %s", strings.Join(diff, ",")))}
```

Comment:
We could be more concrete to avoid confusion. It is forbidden to update the Shoot spec only - metadata and status can be updated.

```suggestion
			return field.ErrorList{field.Forbidden(fldPath, fmt.Sprintf("cannot update shoot spec if deletion timestamp is set. Requested changes: %s", strings.Join(diff, ",")))}
```

Reply:
This is better error message.